### DISCUSSION TITLE
No enums are accessible

### DISCUSSION THREAD

**Discussion by Nowaha** [Minestom/Minestom#2361] [Help] [answered]
All of my Minestom enums are messed up for some unknown reason.
![image](https://github.com/user-attachments/assets/2f8c1b9e-2ab3-43cc-b11d-38e4a6f8ae63)

They are unusable in IntelliJ, but they do work when building.
![image](https://github.com/user-attachments/assets/5987aa24-9dda-4fcf-b3a2-da68db4f779d)
![image](https://github.com/user-attachments/assets/7a1585d5-516c-48e7-af39-fc487362fd86)

I tried to
- Invalidate Caches & Restart
- Delete `.gradle` and `.idea` and `build` and reimporting
- Create a brand new project
- Using a different snapshot (now on the current latest, `789befee31`)

I'm using Kotlin, Java 21 and Gradle 8.10

This is the same problem as #2215, but the solution(?) given there does nothing for me.

This is my `build.gradle.kts`:
```kotlin
plugins {
    kotlin("jvm") version "1.9.22"
    id("com.gradleup.shadow") version "8.3.0"
}

java {
    toolchain {
        languageVersion = JavaLanguageVersion.of(21)
    }
}

kotlin {
    jvmToolchain(21)
}

group = "xyz.lyulf"
version = "1.0-SNAPSHOT"

repositories {
    mavenCentral()
}

dependencies {
    implementation("net.minestom:minestom-snapshots:789befee31")
}

tasks {
    jar {
        manifest {
            attributes["Main-Class"] = "xyz.lyulf.minestomplayground.Main"
        }
    }

    build {
        dependsOn(shadowJar)
    }

    shadowJar {
        mergeServiceFiles()
        archiveClassifier.set("")
    }
}
```

**Comment by cosrnic** [comment]
What version of intellij are you using?

**Reply by Nowaha** [reply]
I'm using `2023.1.2`

**Reply by cosrnic** [reply, marked as answer]
Try updating to a 2024.x version, that version appears to be released before java 21 was released

**Reply by Nowaha** [reply]
That seems to have fixed the issue, thanks!
